examples/nlp/gpt2_text_generation_with_kerasnlp.py

```diff
@@ -0,0 +1,363 @@
+"""
+Title: GPT2 Text Generation with KerasNLP
+Author: Chen Qian
+Date created: 04/17/2023
+Last modified: 04/17/2023
+Description: Use KerasNLP GPT2 model and `samplers` to do text generation.
+Accelerator: GPU
+"""
+"""
+
+In this tutorial, you will learn to use [KerasNLP](https://keras.io/keras_nlp/) to load a
+pre-trained Large Language Model (LLM) - [GPT-2 model](https://openai.com/research/better-language-models)
+(originally invented by OpenAI), finetune it to a specific text style, and
+generate text based on users' input (also known as prompt). You will also learn
+how GPT2 adapts quickly to non-English languages, such as Chinese.
+"""
+
+"""
+##  Before we begin
+
+Colab offers different kinds of runtimes. Make sure to go to **Runtime ->
+Change runtime type** and choose the GPU Hardware Accelerator runtime
+(which should have >12G host RAM and ~15G GPU RAM) since you will finetune the
+GPT-2 model. Running this tutorial on CPU runtime will take hours.
+"""
+
+"""
+## Install KerasNLP and Import Dependencies.
+"""
+
+"""shell
+pip install -q -U git+https://github.com/keras-team/keras-nlp.git@master
+"""
+
+import keras_nlp
+import tensorflow as tf
+from tensorflow import keras
+import time
+
+"""
+## Introduction to Generative Large Language Models (LLMs)
+
+Large language models (LLMs) are a type of machine learning models that are
+trained on a large corpus of text data to generate outputs for various natural
+language processing (NLP) tasks, such as text generation, question answering,
+and machine translation.
+
+Generative LLMs are typically based on deep learning neural networks, such as
+the [Transformer architecture](https://arxiv.org/abs/1706.03762) invented by
+Google researchers in 2017, and are trained on massive amounts of text data,
+often involving billions of words. These models, such as Google [LaMDA](https://blog.google/technology/ai/lamda/)
+and [PaLM](https://ai.googleblog.com/2022/04/pathways-language-model-palm-scaling-to.html),
+are trained with a large dataset from various data sources which allows them to
+generate output for many tasks. The core of Generative LLMs is predicting the
+next word in a sentence, often referred as **Causal LM Pretraining**. In this
+way LLMs can generate coherent text based on user prompts. For a more
+pedagogical discussion on language models, you can refer to the
+[Stanford CS324 LLM class](https://stanford-cs324.github.io/winter2022/lectures/introduction/).
+"""
+
+"""
+## Introduction to KerasNLP
+
+Large Language Models are complex to build and expensive to train from scratch.
+Luckily there are pretrained LLMs available for use right away. One toolkit that
+offers state-of-the-art pretrained models for free is [KerasNLP](https://keras.io/keras_nlp/).
```

Comment:
"for free" kinda makes us feel like too much of a salesman here for my taste. I would probably just say

KerasNLP provides a large number of pre-trained checkpoints that allow you to experiment with SOTA models without needing to train them yourself.

or something like that